queries.sql

```diff
@@ -23,3 +23,63 @@ SELECT * FROM animals WHERE name != 'Gabumon';
 
 /*Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)*/
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+/* Inside a transaction update the animals table by setting the species column
+ to unspecified. Verify that change was made. 
+ Then roll back the change and verify that the species 
+ columns went back to the state before the transaction. */
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+/* Update the animals table by setting the species 
+   column to digimon for all animals that have a 
+   name ending in mon. */
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals;
+
+/* Delete all records in the animals table, then roll back the transaction.
+verify if all records in the animals table still exists. */
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals
+
+/* Delete all animals born after Jan 1st, 2022.
+Create a savepoint for the transaction.
+Update all animals' weight to be their weight multiplied by -1.
+Rollback to the savepoint
+Update all animals' weights that are negative to be their weight multiplied by -1.
+Commit transaction */
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT update_weight;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO update_weight;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+-- QUERIES
+
+--How many animals are there?
+SELECT COUNT(*) FROM animals;
+-- How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+--What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+--Who escapes the most, neutered or not neutered animals?
+SELECT neutered, SUM(escape_attemps) FROM animals GROUP BY (neutered);
```

Comment:
This query should have to answer the question of `Who escapes the most, neutered or not neutered animals?` so it should have to return the maximum count for either neutered or not neutered animals. Hint: instead of `SUM` consider using `MAX`

Reply:
Hi @Mre55, I did the change requested